tests/cases/fourslash/codeFixMissingTypeAnnotationOnExports50-generics-with-default.ts

```diff
@@ -0,0 +1,17 @@
+/// <reference path='fourslash.ts'/>
+
+// @isolatedDeclarations: true
+// @declaration: true
+// @file lib.d.ts
+////interface MyIterator<T, TReturn = any, TNext = undefined> {}
+////let x: MyIterator<number>;
+////export const y = x;
+
+verify.codeFix({
+    description: "Add annotation of type 'MyIterator<number>'",
+    index: 0,
+    newFileContent:
+`interface MyIterator<T, TReturn = any, TNext = undefined> {}
```

Comment:
Do we have tests for generators elsewhere? I seem to recall us auto-adding type long type annotations for those...

Reply:
Sure, I can add a case with Generator.

For reasons I don't fully understand, `Generator<T>` and `Iterator<T>` get generated with the full 3 type arguments, whereas `Iterable<T>` and `IterableIterator<T>` are special cased inside the TypeChecker to omit the extra arguments when unnecessary.

The special casing inside `typeReferenceToTypeNode`:
```
                            // Maybe we should do this for more types, but for now we only elide type arguments that are
                            // identical to their associated type parameters' defaults for `Iterable`, `IterableIterator`,
                            // `AsyncIterable`, and `AsyncIterableIterator` to provide backwards-compatible .d.ts emit due
                            // to each now having three type parameters instead of only one.
                            if (
                                isReferenceToType(type, getGlobalIterableType(/*reportErrors*/ false)) ||
                                isReferenceToType(type, getGlobalIterableIteratorType(/*reportErrors*/ false)) ||
                                isReferenceToType(type, getGlobalAsyncIterableType(/*reportErrors*/ false)) ||
                                isReferenceToType(type, getGlobalAsyncIterableIteratorType(/*reportErrors*/ false))
                            ) {
                                if (
                                    !type.node || !isTypeReferenceNode(type.node) || !type.node.typeArguments ||
                                    type.node.typeArguments.length < typeParameterCount
                                ) {
                                    while (typeParameterCount > 0) {
                                        const typeArgument = typeArguments[typeParameterCount - 1];
                                        const typeParameter = type.target.typeParameters[typeParameterCount - 1];
                                        const defaultType = getDefaultFromTypeParameter(typeParameter);
                                        if (!defaultType || !isTypeIdenticalTo(typeArgument, defaultType)) {
                                            break;
                                        }
                                        typeParameterCount--;
                                    }
                                }
                            }
```

Reply:
That was a part of #58243.

Reply:
Makes sense. Thanks for the context.